config: add --k8s-service-cidr config option and deprecate --service-cluster-ip-range

dcbw opened the pull request:
The two values do the same thing, and --service-cluster-ip-range wins if present
to maintain backwards compatibility. But there's now config file support for
service-cidr and it now has a default too.

@shettyg @girishmg @pecameron @JacobTanenbaum @danwinship @squeed

danwinship commented on go-controller/pkg/config/config_test.go:
```diff
@@ -513,6 +513,20 @@ service-cidr=172.18.0.0/24
 		Expect(err).NotTo(HaveOccurred())
 	})
 
+	It("invalid k8s-service-cidr results in an error", func() {
```
(I don't know if the existing tests use this consistently, but note that `It()` actually prints "`It `" before the test description and so expects a description that completes a sentence, like `"throws an error if k8s-service-cidr is invalid"`. You can use `Specify()` instead of `It()` if you don't want that.)

dcbw replied:
Fixed.

danwinship commented on docs/ovnkube.1:
```diff
@@ -16,10 +16,10 @@ IP address/subnet mask/hostsubnetlength, the hostsubnetlength is optional and if
 A CIDR notation IP range from which k8s assigns service cluster IPs.
 This should be the same as the one provided for kube-apiserver's
-\fB\--service-cluster-ip-range\fR option.
+\fB\--k8s-service-cidr\fR option.
```
don't replace this one; it's referring to kube-apiserver's flag, not ovnkube's

dcbw replied:
thanks, fixed this and another spot I made this mistake.

shettyg commented:
We are doing this so that ovn-kube has the same options as k8s-apiserver? Or is there more to it?

dcbw commented:
> We are doing this so that ovn-kube has the same options as k8s-apiserver? Or is there more to it?

@shettyg first I couldn't think of a reason why --service-cluster-ip-range should *not* be given, but some code allowed for it to be blank. So I cleaned that up and added a default, but that means it goes into pkg/config/config.go.

And *that* made me think that since it's a kubernetes option, it should be grouped with the other k8s-* options but it didn't have that prefix. So then I made that change, shortened the name to one we were already using elsewhere ("service-cidr"), and thus the PR was born...

It's basically about naming consistency with other k8s related options, adding defaults for the option, and making it required.

shettyg commented:
The name `--service-cluster-ip-range` mainly came because kube-apiserver takes the exact same name and I thought people will be familiar with it.

But, I am okay with this.

dcbw commented:
> The name `--service-cluster-ip-range` mainly came because kube-apiserver takes the exact same name and I thought people will be familiar with it.

@shettyg I agree, so I tried to keep it as a CLI-only backwards compat option. I don't think it should go away for a while, if ever.
